skaffold.md

```diff
@@ -0,0 +1,297 @@
+
+## Operating modes
+
+Skaffold has 2 operating modes: `skaffold dev` and `skaffold run`.
+
+### skaffold dev
+
+Updates your deployed application continually:
+
+* Watches your source code and the dependencies of your docker images for changes and runs a build and deploy when changes are detected
+* Streams the logs from deployed containers to your terminal
+* Creates a continuous build-deploy loop that only warn on errors
+
+### skaffold run
+
+Runs a Skaffold pipeline once, exits on any errors in the pipeline. Use for:
+
+* Continuous integration or continuous deployment pipelines
+* Sanity checking after iterating on your application
+
+## Pluggability
+
+Skaffold has a pluggable architecture that allows you to choose which tools you want to use.  In the following exercise, you will build the container locally using the Docker daemon, push it to an ECR registry, and deploy it to EKS by applying a Kubernetes manifest. 
+
+![plugabbility](https://github.com/jicowan/aws-microservices-deploy-options/blob/master/images/skaffold.jpg)
+
+## Prerequisites
+
+### Install ECR Credential Helper
+
+The ECR credential helper makes it easier to use ECR by seamless passing your AWS credentials to the service.  When you use the credential helper there is no need to use `docker login` or `docker logout`.  
+
+> If you're planning to use the credential helper with an assumed role, you'll need to set the environment variable `AWS_SDK_LOAD_CONFIG=true`.  The IAM principle you're using has to have permission to ECR too. 
+
+Install the credential helper with  `go get`:
+
+```
+`go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login`
+```
+
+Place the `docker-credential-ecr-login` binary on your `PATH` and add the following to the contents of your `~/.docker/config.json` file: 
+
+```
+{
+    "credHelpers": {
+        "[account_number].dkr.ecr.[region].amazonaws.com>": "ecr-login" 
+        }
+}
+```
+
+This configures the daemon to use the credential helper for a specific ECR registry. 
+
+> Configuring a specific registry will only work with Docker v1.13.0 or higher.
+
+### Provision an EKS Cluster
+
+Download and install the latest release of [eksctl](https://eksctl.io/) from Weave. 
+
+```
+`curl --silent --location "https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_$(uname -s)_amd64.tar.gz" | tar xz -C /tmp
+sudo mv /tmp/eksctl /usr/local/bin`
+```
+
+To create a basic cluster, run: 
+
+```
+eksctl create cluster
+```
+
+This will create a cluster in a new VPC in the `us-west-2` region with the following defaults: 
+
+* Two (2) `m5.large` nodes
+* Nodes are built from EKS optimized AMI
+
+### Create an ECR registry
+
+```
+REPOSITORY_URI=$(aws ecr create-repository --repository-name k8s-skaffold/skaffold-example | jq -r 'repository.repositoryUri')  
```

Comment:
Break the command over multiple lines, separated by back slash, for readability?